### PR TITLE
[1LP][RFR] Fixing rest_mapping related TC for OSP

### PR DESCRIPTION
**Pull request by mnadeem92**
Signed-off-by: mnadeem92 <mnadeem@redhat.com>

 {{ pytest: cfme/tests/v2v/test_rest_migrations.py  -k "test_rest_mapping_create"  --use-provider={osp13-ims,rhv-ims}  --use-provider vsphere67-ims --provider-limit 2 -v }}

5.11 - Passed
5.10 - Failed due to TimedOutError at fixture function( _start_event_workers_for_osp) which need to address in a separate PR.

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit b59f763693d42f44417c0a90647ac2c695627507

The *local* fixture **`get_clusters`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

The *local* fixture **`get_datastores`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

The *local* fixture **`get_networks`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by sshveta** [comment]
Please address one failure in 5.10

**Comment by mnadeem92** [comment]
@sshveta  The failure on 5.10 is at the fixture level(  _start_event_workers_for_osp ) that need to debug and address in a separate PR, As the changes in this PR is not directly related with this issue. IMO lets do not hold this PR as it is passing with 5.11 and we need it for upcoming 5.11.5 build and upstream testing now.

File: cfme/tests/v2v/test_rest_migrations.py [hunk 1]
```diff
@@ -42,8 +42,13 @@ def get_clusters(appliance, provider, source_provider):
         target_cluster = provider.data.get("clusters")[0]
     except IndexError:
         pytest.skip("Cluster not found in given provider data")
+
+    clusters_obj = appliance.rest_api.collections.clusters.all
+    if provider.one_of(OpenStackProvider):
+        clusters_obj += appliance.rest_api.collections.cloud_tenants.all
```
Comment (digitronik) [inline cfme/tests/v2v/test_rest_migrations.py]: optional: I would use `extend`

File: cfme/tests/v2v/test_rest_migrations.py [hunk 2]
```diff
@@ -42,8 +42,13 @@ def get_clusters(appliance, provider, source_provider):
         target_cluster = provider.data.get("clusters")[0]
     except IndexError:
         pytest.skip("Cluster not found in given provider data")
+
+    clusters_obj = appliance.rest_api.collections.clusters.all
+    if provider.one_of(OpenStackProvider):
```
Comment (digitronik) [inline cfme/tests/v2v/test_rest_migrations.py]: Required: Please Add one `comment` explaining why we combining clusters and cloud tenants for OpenStack provider.

File: cfme/tests/v2v/test_rest_migrations.py [hunk 3]
```diff
@@ -65,15 +70,23 @@ def get_clusters(appliance, provider, source_provider):
 @pytest.fixture(scope="function")
 def get_datastores(appliance, provider, source_provider):
     datastores = {}
+    datastores_obj = appliance.rest_api.collections.data_stores.all
+    if provider.one_of(OpenStackProvider):
```
Comment (digitronik) [inline cfme/tests/v2v/test_rest_migrations.py]: ^^ same

**Comment by dajoRH** [comment]
I detected some fixture changes in commit a7c4a5e61512dc504af839151bbb3c6909651e90

The *local* fixture **`get_clusters`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

The *local* fixture **`get_datastores`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

The *local* fixture **`get_networks`** is used in the following files:
- **cfme/tests/v2v/test_rest_migrations.py**
    - *test_rest_mapping_create*
    - *test_rest_plan_create*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Pull request merged**